src/main/java/edu/illinois/library/cantaloupe/image/Metadata.java

```diff
@@ -155,9 +153,27 @@ public Orientation getOrientation() {
     }
 
     private void readOrientationFromEXIF() {
+        Field field = exif.getField(Tag.ORIENTATION);
         Object value = exif.getValue(Tag.ORIENTATION);
-        if (value != null) {
-            orientation = Orientation.forEXIFOrientation((int) value);
+        if (field != null && value != null) {
+            switch (field.getDataType()) {
+              case LONG:
+              case SLONG:
+              case SHORT:
+              case SSHORT:
+                // According to spec the orientation must be an unsigned short (16 bit)
+                // However, we have seen exif data in the wild with LONG and SLONG types
+                // Thus to be lenient we accept either and convert to int (github issue #548)
+                if (value instanceof Long) {
```

Comment:
Alternatively we could skip the `field.getDataType()` check and only do the instanceof check...